encoding/yaml/error/YAMLError.ts

```diff
@@ -0,0 +1,17 @@
+import { Mark } from "../Mark.ts";
+
+const { DenoError, ErrorKind } = Deno;
+
+export class YAMLError extends DenoError<typeof ErrorKind.Other> {
+  constructor(
+    message = "(unknown reason)",
+    protected mark: Mark | string = ""
+  ) {
+    super(ErrorKind.Other, `${message} ${mark}`);
+    this.name = this.constructor.name;
+  }
+
+  public toString(_compact: boolean): string {
+    return `${this.name}: ${this.message} ${this.mark}`;
+  }
+}
```

Comment:
Since there's only one error type, it seems superfluous to create `error` directory. How about `yaml/error.ts`?

Reply:
Was an anticipation made at the beginning of the port.

Done.